File: BINARY-SEARCH/bs.java
class Solution {
    public int search(int[] nums, int target) {
        int end = nums.length - 1;
        int mid = end / 2;
        int start = 0;
         while (start <= end){
            if(mid < target){
                start = mid + 1;
                
            }
            else if(mid > target){
                end = mid - 1;
            }
            else{
                return mid;
            }
         }
        return -1;
    }
}

File: BINARY-SEARCH/bs.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        start = 0
        end = len(nums) - 1
        mid = len(nums) / 2
        while start <= end:
            if nums[mid] < target:
                start = mid + 1
            elif nums[mid] > target:
                end = mid - 1
            else:
                return mid
            mid = (start + end) // 2
        return -1